inject/src/main/java/io/avaje/inject/spi/DBeanMap.java

```diff
@@ -6,21 +6,17 @@
 
 import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
-import java.util.HashSet;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * Map of types (class types, interfaces and annotations) to a DContextEntry where the
  * entry holds a list of bean instances for that type.
  */
 final class DBeanMap {
   private static final Optional<Object> EMPTY = Optional.empty();
-  private final Map<String, DContextEntry> beans = new LinkedHashMap<>();
-  private final Set<String> qualifiers = new HashSet<>();
+  private final Map<String, DContextEntry> beans = new ConcurrentHashMap<>();
+  private final Set<String> qualifiers = Collections.synchronizedSet(new HashSet<>());
```

Comment:
Do we require synchronization? The async beans don't have an opportunity to write to these collections. They are registered as jakarta providers.